src/app/features/swaps/shared/components/assets-selector/services/assets-selector-service/assets-selector.service.ts

```diff
@@ -56,12 +62,46 @@ export class AssetsSelectorService {
     this.subscribeOnAssetChange();
   }
 
+  private isSupportedOnChainNetwork(
+    blockchain: BlockchainName
+  ): blockchain is SupportedOnChainNetworks {
+    return Object.entries(INSTANT_TRADE_PROVIDERS).some(
+      ([supportedNetwork, providers]) => supportedNetwork === blockchain && providers.length > 0
+    );
+  }
+
+  private isUserFirstNetworkSelection(
+    fromBlockchain: BlockchainName,
+    assetTypeKey: 'fromAssetType' | 'toBlockchain'
+  ): boolean | string {
+    return (
+      !fromBlockchain ||
+      assetTypeKey !== 'toBlockchain' ||
+      (assetTypeKey === 'toBlockchain' &&
+        fromBlockchain &&
+        this.swapFormService.inputValue.toBlockchain) ||
```

Comment:
Boolean(assetTypeKey === 'toBlockchain' &&
        fromBlockchain &&
        this.swapFormService.inputValue.toBlockchain)